provider/transformation.go

```diff
@@ -0,0 +1,209 @@
+package provider
+
+import (
+	"fmt"
+)
+
+type (
+	OutTransformationProvider[T any] interface {
+		convertToInt(T) (int64, error)
+		convertToString(T) (string, error)
+		convertToFloat(T) (float64, error)
+		convertToBool(T) (bool, error)
+		outTransformations() []OutTransformation
+	}
+	InTransformationProvider interface {
+		setParam(string, any) error
+		inTransformations() []InTransformation
+	}
+	ScriptTransformationProvider[A any] interface {
```

Comment:
Unused?

Reply:
I think it was used somewhere in between, but isn't anymore, will remove it